### PR TITLE
reflector: Implement gRPC server reflection

### PR DESCRIPTION
**Pull request by juliaogris**
Implement gRPC schema reflection in the SchemaReflector class.
Generator gRPC and Proto Kotlin stubs with wire from
grpc/reflection/v1alpha/reflection.proto

Outstanding work (marked with `// TODO`):

* Return correct error codes (`grpc::NOT_FOUND `mostly)
* Implement `all_extension_numbers_of_type(typeName)`
* Implement `file_containing_extension(typeName, extNum)`

### CODE REVIEW

File: wire-library/wire-reflector/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=wire-reflector
+POM_NAME=Wire
+POM_DESCRIPTION=Reflector
```
Comment (oldergod) [inline wire-library/wire-reflector/gradle.properties:3]: Maybe something more speaking? It can be a sentence.

**Review by oldergod — COMMENTED** [review]
When ready for review, splitting the generation vs the manually written code into 2 commits would help a lot.
In some files as well (SchemaReflector.kt), the indentation seems much bigger than what other files use. You also need copyright header in the new files.

File: wire-library/wire-reflector/gradle.properties (hunk 1 above)
Comment (juliaogris) [reply wire-library/wire-reflector/gradle.properties]: Updated.

File: wire-library/wire-reflector/src/jvmMain/kotlin/com/squareup/wire/reflector/SchemaReflector.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import grpc.reflection.v1alpha.ErrorResponse
+import grpc.reflection.v1alpha.FileDescriptorResponse
+import grpc.reflection.v1alpha.ListServiceResponse
+import grpc.reflection.v1alpha.ServerReflectionRequest
+import grpc.reflection.v1alpha.ServerReflectionResponse
+import grpc.reflection.v1alpha.ServiceResponse
+import okio.ByteString
+
+/**
+ * This converts a Wire Schema model to a protobuf DescriptorProtos model and serves that.
```
Comment (swankjesse) [inline wire-library/wire-reflector/src/jvmMain/kotlin/com/squareup/wire/reflector/SchemaReflector.kt:31]: This comment is less fitting now that the type conversion happens elsewhere.

**Review by swankjesse — APPROVED** [review]
Love it.

File: wire-library/wire-reflector/src/jvmMain/kotlin/com/squareup/wire/reflector/SchemaReflector.kt (hunk 2 above)
Comment (juliaogris) [reply wire-library/wire-reflector/src/jvmMain/kotlin/com/squareup/wire/reflector/SchemaReflector.kt:31]: ok, I'll fix this in the next pr.

**Pull request merged**